content/en/developers/dogstatsd/unix_socket.md

```diff
@@ -16,161 +16,222 @@ further_reading:
+Instead of using an `IP:port` pair to establish connections, Unix Domain Sockets use a placeholder socket file. Once the connection is open, data is transmitted in the same [datagram format][1] as for the UDP transport. When the Agent restarts, the existing socket is deleted and replaced by a new one. Client libraries detect this change and connect seamlessly to the new socket.
 
 **Note:** By design, UDS traffic is local to the host, which means the Datadog Agent must run on every host you send metrics from.
 
 ## Setup
 
-To set up DogStatsD, enable the DogStatsD server through the `dogstatsd_socket` parameter. Then, configure the DogStatsD client in your code.
+To set up DogStatsD with Unix Domain Socket , enable the DogStatsD server through the `dogstatsd_socket` parameter. Then, configure the [DogStatsD client in your code](#dogstatsd-client-configuration).
 
-### Agent
+To enable the Agent DogStatsD UDS:
 
-To enable the Agent DogStatsD server:
+{{< tabs >}}
+{{% tab "Host" %}}
 
-1. Edit the [Agent's main configuration file][2] to set `dogstatsd_socket` to the path where DogStatsD should create its listening socket:
+1. Edit the [Agent's main configuration file][1] to set `dogstatsd_socket` to the path where DogStatsD should create its listening socket:
 
     ```yaml
     ## @param dogstatsd_socket - string - optional - default: ""
-    ## Listen for Dogstatsd metrics on a Unix Socket (*nix only). Set to a valid filesystem path to enable.
+    ## Listen for Dogstatsd metrics on a Unix Socket (*nix only).
+    ## Set to a valid filesystem path to enable.
     #
     dogstatsd_socket: '/var/run/datadog/dsd.socket'
     ```
 
-2. [Restart your Agent][3].
+2. [Restart your Agent][2].
 
-**Note**: You can also set the socket path with the `DD_DOGSTATSD_SOCKET` environment variable for the container Agent.
 
-### DogStatsD Client
+[1]: /agent/guide/agent-configuration-files/#agent-main-configuration-file
+[2]: /agent/guide/agent-commands
+{{% /tab %}}
+{{% tab "Docker" %}}
 
-#### Native support in client libraries
+1. Set the socket path with the `DD_DOGSTATSD_SOCKET=<YOUR_UDS_PATH>` environment variable on the Agent container.
 
-The following official DogStatsD client libraries natively support UDS traffic. Refer to the library's documentation on how to enable UDS traffic. **Note**: As with UDP, enabling client-side buffering is highly recommended to improve performance on heavy traffic:
+2. The socket file then needs to be accessible to the client containers. To achieve this, Datadog recommends mounting a host directory on both sides (read-only in your client containers, read-write in the Agent container). Mounting the parent folder instead of the individual socket enables socket communication to persist across DogStatsD restarts:
```

Comment:
```suggestion
2. Make the socket file accessible to the client containers by mounting a host directory on both sides (read-only in your client containers and read-write in the Agent container). Mounting the parent folder instead of the individual socket enables socket communication to persist across DogStatsD restarts:
```